antarest/study/storage/rawstudy/raw_study_service.py

```diff
@@ -402,11 +402,38 @@ def unarchive(self, study: RawStudy) -> None:
         Raises:
             BadArchiveContent: If the archive is corrupted or in an unknown format.
         """
-        with open(self.get_archive_path(study), mode="rb") as fh:
+        with open(self.find_archive_path(study), mode="rb") as fh:
             self.import_study(study, fh)
 
-    def get_archive_path(self, study: RawStudy) -> Path:
-        return Path(self.config.storage.archive_dir / f"{study.id}.zip")
+    def find_archive_path(self, study: RawStudy) -> Path:
+        """
+        Fetch for archive path of a study if it exists else raise an incorrectly archived study.
+
+        Args:
+            study: The study to get the archive path for.
+
+        Returns:
+            The full path of the archive file (zip or 7z).
+        """
+        archive_dir: Path = self.config.storage.archive_dir
+        for suffix in [".zip", ".7z"]:
+            path = archive_dir.joinpath(f"{study.id}{suffix}")
+            if path.is_file():
+                return path
+        raise FileNotFoundError(f"Study {study.id} archiving process is corrupted (no archive file found).")
+
+    def create_archive_path(self, study: RawStudy) -> Path:
```

Comment:
I don't see the point of this method. You could use find_archive_path instead I believe and it would be clearer as you don't really create a path inside this method. If so, you can remove this method

Reply:
This is how Laurent wanted it to be called ... but I would agree with you